src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs

```diff
@@ -50,29 +51,25 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
             var packageMaps = currentTFM.Any(c => c.IsFramework) ? allPackageMaps.Where(x => x.NetCorePackagesWorkOnNetFx).ToArray<NuGetPackageMap>() : allPackageMaps;
             var references = await project.GetNuGetReferencesAsync(token).ConfigureAwait(false);
 
-            foreach (var packageReference in references.PackageReferences.Where(r => !state.PackagesToRemove.Contains(r)))
+            foreach (var packageReference in state.Packages)
             {
                 foreach (var map in packageMaps.Where(m => ContainsPackageReference(m.NetFrameworkPackages, packageReference.Name, packageReference.Version)))
                 {
-                    state.PossibleBreakingChangeRecommended = true;
                     _logger.LogInformation("Marking package {PackageName} for removal based on package mapping configuration {PackageMapSet}", packageReference.Name, map.PackageSetName);
-                    state.PackagesToRemove.Add(packageReference);
+                    state.Packages.Remove(packageReference, BuildBreakRisk.Medium);
```

Comment:
Not something that needs addressed in this PR, but in the future it might be nice to allow package maps to specify the `BuildBreakRisk` that makes sense for them. Probably best to wait for a customer to ask for that first, though, to avoid needless complexity.

Reply:
Great idea! I created an issue to track that.